griptape/tools/google_drive/tool.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+import logging
+from schema import Schema, Literal, Optional
+from attr import define
+from griptape.artifacts import TextArtifact, ErrorArtifact, InfoArtifact, ListArtifact
+from griptape.utils.decorators import activity
+from griptape.tools import BaseGoogleClient
+from googleapiclient.errors import HttpError
+from google.auth.exceptions import MalformedError
+
+
+@define
+class GoogleDriveClient(BaseGoogleClient):
+    LIST_FILES_SCOPES = ['https://www.googleapis.com/auth/drive.readonly']
+    UPLOAD_FILE_SCOPES = ['https://www.googleapis.com/auth/drive.file']
+
+    @activity(config={
+        "description": "Can be used to list files from Google Drive",
+        "schema": Schema({
+            Literal(
+                "drive_owner_email",
+                description="email of the Google Drive's owner"
+            ): str,
+            Optional(Literal(
+                "max_files",
+                description="maximum number of files to return"
+            )): int
+        })
+    })
+    def _build_client(self, scopes: list[str], values: dict) -> object:
+        from google.oauth2 import service_account
+        from googleapiclient.discovery import build
+        credentials = service_account.Credentials.from_service_account_info(
+            self.service_account_credentials, scopes=scopes
+        )
+        delegated_credentials = credentials.with_subject(values["drive_owner_email"])
+        service = build('drive', 'v3', credentials=delegated_credentials)
+        return service
+
+    def list_files(self, params: dict) -> ListArtifact | ErrorArtifact:
+
+        values = params["values"]
+
+        try:
+            service = self._build_client(self.LIST_FILES_SCOPES, values)
+            results = service.files().list(pageSize=values.get('max_files', 10)).execute()
+            items = results.get('files', [])
+            return items
+
+        except Exception as e:
+            logging.error(e)
+            return ErrorArtifact(f"error retrieving files from Drive: {e}")
+
+    @activity(config={
+        "description": "Can be used to upload a file to Google Drive",
+        "schema": Schema({
+            Literal(
+                "drive_owner_email",
+                description="email of the Google Drive's owner"
+            ): str,
+            Literal(
+                "file_path",
+                description="path to the file to be uploaded"
+            ): str,
+            Literal(
+                "file_name",
+                description="name to give the file on Google Drive"
+            ): str,
+            Literal(
+                "mime_type",
```

Comment:
Is this a requirement to have this field defined? Ideally, the LLM shouldn't know about this.